app/src/main/java/fr/free/nrw/commons/upload/UploadModel.java

```diff
@@ -369,6 +406,10 @@ public boolean equals(@Nullable Object obj) {
         public int hashCode() {
             return super.hashCode();
         }
+
+        public String getFileName() {
```

Comment:
javadoc:
Choose a filename for the media.
Currently, the caption is used as a filename. If several languages have been entered, the first language is used.

Reply:
done